tools/snp.sh

```diff
@@ -985,17 +1118,28 @@ main() {
       ;;
 
     setup-host)
-      install_dependencies
+      if [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; then
+        ubuntu_install_dependencies
+      else
+        rhel_install_dependencies
+      fi
+
+      install_nasm_from_source
```

Comment:
Move all of this logic into a method `install_dependencies`. The previous code reads almost like a book what is happening. The above and below OS checks make it cluttered and hard to understand. This is the main method. Make the code as simple, clean and easy to read. That is the goal.